src/apps/frontend/pages/authentication/login/login-form.hook.ts

```diff
@@ -0,0 +1,49 @@
+import { useFormik } from 'formik';
+import { useCallback } from 'react';
+import * as Yup from 'yup';
+
+import constant from '../../../constants';
+import { useAuthContext } from '../../../contexts';
+import useAsync from '../../../contexts/async.hook';
+
+const useLoginForm = () => {
+  const { login: signIn } = useAuthContext();
+  const {
+    status, error, result, apiCallback,
+  } = useAsync();
+
```

Comment:
This semantic seems a bit odd.

I would expect it to be:

```
const {
    status, error, result, asyncOperation: as login
  } = useAsync(signIn(username, password)));
```

This should not be needed:

```
const login = useCallback(
    ({ username, password }) => {
      apiCallback(() => signIn(username, password));
    },
    [apiCallback, signIn],
  );
```

Reply:
refactored it.